### PR TITLE
ci: Fix ace editor E2E typing flakes

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### What does this PR accomplish?
It fixes some of the most recent E2E typing flakes that seem to be more exacerbated in combination with the ace_editor.

We are using the UI flow to start a new question or a model in way too many places. This PR creates two new helpers that will open the new native question/model page programmatically by providing the base64 encoded query.

**It is possible to pre-select a database and a query, if needed.**

> [!Important]
> These new helpers *do not* return the ace editor on purpose so that you can't chain `.type()` on it!

**Before** ❌ 
```js
startNewNativeQuestion().type("select 1");
```

**Now** ✅ 
```js
startNewNativeQuestion();
focusNativeEditor().type("select 1");
```

### Next steps
I plan to start applying these new helpers to a lot of other tests that haven't been flaky to begin with. For now it's important to keep the PR reasonably small to minimize the chances of merge conflicts.

The new helper I'm thinking about:
```ts
startNewNativeQuery();
enterNativeQuery(query: string, config: { runQuery: boolean, delay: number});
```

### CODE REVIEW

File: e2e/test/scenarios/visualizations-tabular/table.cy.spec.js [hunk 1]
```diff
@@ -291,15 +292,19 @@ describe("scenarios > visualizations > table", () => {
...
 
-    hovercard().within(() => {
-      cy.contains("No special type");
-      cy.findByText("No description");
-    });
+    cy.log("Assert");
+    cy.findAllByTestId("header-cell").filter(":contains(CATEGORY)").realHover();
+    hovercard()
+      .should("contain", "No special type")
+      .and("contain", "No description");
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/visualizations-tabular/table.cy.spec.js:307]: ### Stress-test
100/100 ✅ https://github.com/metabase/metabase/actions/runs/10077085342

File: e2e/test/scenarios/native/native-reproductions.cy.spec.js [hunk 2]
```diff
@@ -556,18 +557,15 @@ describe("issue 30680", () => {
     cy.signInAsAdmin();
   });
 
-  it("should not render native editor buttons when 'Metadata' tab is open", () => {
-    cy.visit("/model/new");
+  it("should not render native editor buttons when 'Metadata' tab is open (metabase#30680)", () => {
+    startNewNativeModel({ query: "select 1" });
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/native/native-reproductions.cy.spec.js:561]: ### Stress-test
50x https://github.com/metabase/metabase/actions/runs/10078207336

File: e2e/test/scenarios/models/reproductions.cy.spec.js [hunk 3]
```diff
@@ -1668,12 +1669,8 @@ describe("issue 37009", () => {
   });
 
   it("should prevent saving new and updating existing models without result_metadata (metabase#37009)", () => {
-    cy.visit("/model/new");
-    cy.findByTestId("new-model-options")
-      .findByText("Use a native query")
-      .click();
+    startNewNativeModel({ query: "select * from products" });
 
-    focusNativeEditor().type("select * from products");
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/models/reproductions.cy.spec.js:1676]: ### Stress-test
50x https://github.com/metabase/metabase/actions/runs/10078243126

File: e2e/test/scenarios/native/native_subquery.cy.spec.js [hunk 4]
```diff
@@ -311,15 +314,15 @@ describe("scenarios > question > native subquery", () => {
...
     };
 
-    cy.createNativeQuestion(questionDetails).then(
+    createNativeQuestion(questionDetails).then(
       ({ body: { id: baseQuestionId } }) => {
         const tagID = `#${baseQuestionId}`;
 
-        startNewNativeQuestion().type(`SELECT * FROM {{${tagID}`);
+        startNewNativeQuestion();
+        focusNativeEditor().type(`SELECT * FROM {{${tagID}`);
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/native/native_subquery.cy.spec.js:322]: ### Stress-test
50x https://github.com/metabase/metabase/actions/runs/10078428942

File: e2e/test/scenarios/native/native_subquery.cy.spec.js [hunk 5]
```diff
@@ -286,21 +289,21 @@ describe("scenarios > question > native subquery", () => {
...
 
-    cy.createQuestion(questionDetails).then(
+    createQuestion(questionDetails).then(
       ({ body: { id: nestedQuestionId } }) => {
         const tagID = `#${nestedQuestionId}`;
         cy.intercept("GET", `/api/card/${nestedQuestionId}`).as("loadQuestion");
 
-        startNewNativeQuestion().type(`SELECT * FROM {{${tagID}`);
+        startNewNativeQuestion();
+        focusNativeEditor().type(`SELECT * FROM {{${tagID}`);
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/native/native_subquery.cy.spec.js:298]: ### Stress-test
50x https://github.com/metabase/metabase/actions/runs/10078434137

**Review by ranquild — APPROVED** [review]

**Pull request merged**